src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -63,30 +68,38 @@ private FieldVariable getFieldVariableForRootTable(String columnName) {
   }
 
   FilterVariable generateFilter(SnapshotBuilderProgramDataRangeCriteria rangeCriteria) {
+    String columnName = getProgramDataOptionColumnName(rangeCriteria.getId());
     return new BooleanAndOrFilterVariable(
         BooleanAndOrFilterVariable.LogicalOperator.AND,
         List.of(
             new BinaryFilterVariable(
-                getFieldVariableForRootTable(rangeCriteria.getName()),
+                getFieldVariableForRootTable(columnName),
                 BinaryFilterVariable.BinaryOperator.GREATER_THAN_OR_EQUAL,
-                new Literal(rangeCriteria.getLow().intValue())),
+                new Literal(rangeCriteria.getLow())),
             new BinaryFilterVariable(
-                getFieldVariableForRootTable(rangeCriteria.getName()),
+                getFieldVariableForRootTable(columnName),
                 BinaryFilterVariable.BinaryOperator.LESS_THAN_OR_EQUAL,
-                new Literal(rangeCriteria.getHigh().intValue()))));
+                new Literal(rangeCriteria.getHigh()))));
   }
 
   FilterVariable generateFilter(SnapshotBuilderProgramDataListCriteria listCriteria) {
     return new FunctionFilterVariable(
         FunctionFilterVariable.FunctionTemplate.IN,
-        getFieldVariableForRootTable(listCriteria.getName()),
-        listCriteria.getValues().stream()
-            .map(value -> new Literal(value.intValue()))
-            .toArray(Literal[]::new));
+        getFieldVariableForRootTable(getProgramDataOptionColumnName(listCriteria.getId())),
+        listCriteria.getValues().stream().map(Literal::new).toArray(Literal[]::new));
+  }
+
+  String getProgramDataOptionColumnName(Integer id) {
```

Comment:
In the interest of making it clear that this is never null, I would use `int` instead.
```suggestion
  String getProgramDataOptionColumnName(int id) {
```